grid-client/graphql/graphql.go

```diff
@@ -9,17 +9,62 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/cenkalti/backoff"
 	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"golang.org/x/exp/rand"
 )
 
 // GraphQl for tf graphql
 type GraphQl struct {
-	url string
+	urls []string
+	r    int
 }
 
 // NewGraphQl new tf graphql
-func NewGraphQl(url string) (GraphQl, error) {
-	return GraphQl{url}, nil
+func NewGraphQl(urls ...string) (GraphQl, error) {
+	if len(urls) == 0 {
+		return GraphQl{}, errors.Errorf("graphql url is required")
+	}
+
+	rand.Shuffle(len(urls), func(i, j int) {
```

Comment:
I wouldn't randomize it, we typically list with the importance (I wouldn't want to go to another stack while the default stack is responding)